jscomp/runtime/caml_int64.ml

```diff
@@ -40,30 +40,21 @@ let of_int32 (lo : nativeint) =
   else {lo ; hi = 0n }
 
 
-let add 
-    ({lo = this_low_; hi = this_high_} : t ) 
-    ({lo = other_low_; hi = other_high_} : t ) = 
-
-  let a48 = shift_right_logical this_high_  16 in
-  let a32 = logand this_high_  0xFFFFn in
-  let a16 = shift_right_logical this_low_  16 in
-  let a00 = logand this_low_  0xFFFFn in
-
-  let b48 = shift_right_logical other_high_  16 in
-  let b32 = logand other_high_   0xFFFFn in
-  let b16 = shift_right_logical other_low_  16 in
-  let b00 = logand other_low_  0xFFFFn in
-
-  begin 
-    let c00 = add a00  b00 in
-    let c16 = add (add (shift_right_logical c00  16) a16)  b16 in
-    let c32 = add (add (shift_right_logical  c16  16) a32)  b32 in
-    let c48 = add (add (shift_right_logical c32 16) a48)  b48 in
-    {lo = 
-       logor (shift_left (logand c16  0xFFFFn)  16)  (logand c00  0xFFFFn); 
-     hi =  logor (shift_left (logand c48 0xFFFFn)  16)  (logand c32  0xFFFFn)
-    }
-  end 
+let add
+    ({lo = this_low_; hi = this_high_} : t)
+    ({lo = other_low_; hi = other_high_} : t) =
+  let low = logand (add this_low_ other_low_) 0xFFFFFFFFn in
+  let overflow = if other_low_ < 1n
+    then sub Nativeint.min_int other_low_ < this_low_
```

Comment:
They should be `-0x80000000` and `0x7fffffff`. It's probably not correct for  bucklescript's `Nativeint` (depending on how we implement it).